src/main/java/io/cdap/plugin/gcp/gcs/StorageClient.java

```diff
@@ -320,17 +321,21 @@ private static String toPath(BlobId blobId) {
   }
 
   public static StorageClient create(String project, @Nullable String serviceAccount,
-                                     Boolean isServiceAccountFilePath) throws IOException {
+                                     Boolean isServiceAccountFilePath, @Nullable Integer readTimeout)
+    throws IOException {
     StorageOptions.Builder builder = StorageOptions.newBuilder().setProjectId(project);
     if (serviceAccount != null) {
       builder.setCredentials(GCPUtils.loadServiceAccountCredentials(serviceAccount, isServiceAccountFilePath));
     }
+    if (readTimeout != null) {
+      builder.setTransportOptions(HttpTransportOptions.newBuilder().setReadTimeout(readTimeout * 1000).build());
+    }
     Storage storage = builder.build().getService();
     return new StorageClient(storage);
   }
 
-  public static StorageClient create(GCPConnectorConfig config) throws IOException {
-    return create(config.getProject(), config.getServiceAccount(), config.isServiceAccountFilePath());
+  public static StorageClient create(GCPConnectorConfig config, Integer readTimeout) throws IOException {
+    return create(config.getProject(), config.getServiceAccount(), config.isServiceAccountFilePath(), readTimeout);
```

Comment:
```
public static StorageClient create(GCPConnectorConfig config) throws IOException {
    return create(config.getProject(), config.getServiceAccount(), config.isServiceAccountFilePath(),  null);
```
This way we will not need to change the underlying classes which are calling it and not using `readTimeout`.

Reply:
Just to clarify, are you proposing to add this additional method along with the existing one?

Reply:
Discussed offline, made the change